docs/spec/governance/software_upgrade.md

```diff
@@ -0,0 +1,133 @@
+## Software Upgrade
+
+Software is upgraded according to a 4-step process.
+
+1. Agree on software upgrade plan via governance (currently via a plaintext proposal).
+2. Once the upgrade plan is accepted, validators each update their software according to proposal from step 1.
+3. Each upgraded validator submits a transaction to the SoftwareUpgradeKeeper.
+4. Once a sufficient quorum of validators has completed step 3, new logic kicks
```

Comment:
what is "sufficient" here? kind of ambiguous should probably have a number of reference to a governance param or something.

Reply:
+1. Something not too close to 2/3rd so we don't risk halting on switch if a couple of validators go of

Reply:
Agreed, also we should clarify a bit how the new logic kicks in - conditionals in the code, for now?